packages/aws-cdk-lib/aws-ecs/lib/log-drivers/aws-log-driver.ts

```diff
@@ -82,6 +82,16 @@ export interface AwsLogDriverProps {
    * @default - AwsLogDriverMode.BLOCKING
    */
   readonly mode?: AwsLogDriverMode;
+
+  /**
+   * When AwsLogDriverMode.NON_BLOCKING is configured, this parameter
+   * controls the size of the non-blocking buffer used to temporarily
+   * store messages. This parameter is not valid with
+   * AwsLogDriverMode.BLOCKING.
+   *
+   * @default - No default
+   */
+  readonly maxBufferSize?: string;
```

Comment:
I think a better way of handling this is make `maxBufferSize` to be a `number` type. This will provide a better user experience as it isn't extremely clear upfront that the user should provide something of the form  1m, 5m, 10m, etc. We can then do the work to append 'm' in the `bind` method.

Reply:
@colifran The problem here is that while "number + m" is the most likely user config, it's not the only thing that Docker supports. The string arg to max-buffer-size can have multiple units:
> format: <number>[<unit>], where unit = b, k, m or g)

So this is valid and works:

```
docker run -it --log-opt mode=non-blocking --log-opt max-buffer-size=4k alpine
```

Reply:
@PettitWesley Ahh I see. In that case, it seems like this property might be a good candidate to utilize an enum-like class. I think we want to help direct the user input and tighten up what a user can provide as an option. As an example,

```ts
export enum BufferSizeType {
  MB = 'MB',
  KB = 'KB',
  GB = 'GB',
}

export class BufferSize {
  public static megabyte(size: number) {
    return new BufferSize(BufferSizeType.MB, size);
  }

  public static kilobyte(size: number) {
    return new BufferSize(BufferSizeType.KB, size);
  }

  public static gigabyte(size: number) {
    return new BufferSize(BufferSizeType.GB, size);
  }
  
  //  more buffer size options

  private constructor(public readonly type: string, public readonly size: number) {}
}
```

What do you think?

Reply:
What does the user input? A string or do they create one of these structs?

Reply:
I think I'd rather not create a new struct just for this field, how about instead I add better validations in the constructor that require that this is a number + some unit character? That would serve the same use case.

Reply:
Docker code:
- https://pkg.go.dev/github.com/docker/go-units#RAMInBytes
- https://github.com/docker/go-units/blob/master/size.go#L36

The code always processes either a string or an int, and uses functions there to convert back and forth.

Reply:
Should this be common validation code that code in a centralized location and not just this log driver?

Reply:
Something like: `IsValidDockerMemoryString()` which I call in the constructor to validate that the user inputted string is valid.

Reply:
As discussed on slack, I will use the existing class:
 https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.Size.html. I think we could just use this as the type for maxBufferSize . I think we could convert to bytes in the bind method. It would looks something like this:
let bufferSize: string | undefined = undefined;
if (props.mode === AwsLogsDriver.NON_BLOCKING) {
  bufferSize = (props.maxBufferSize ?? cdk.Size.mebibytes(1)).toBytes({ rounding: cdk.SizeRoundingBehavior.FLOOR }) + 'b';
}